pkg/cloudprovider/providers/azure/azure_storageaccount.go

```diff
@@ -46,6 +46,9 @@ func (az *Cloud) getStorageAccounts() ([]accountWithLocation, error) {
 			storageType := ""
 			if acct.Sku != nil {
 				storageType = string((*acct.Sku).Name)
+				if matchingAccountType != "" && !strings.EqualFold(matchingAccountType, storageType) {
```

Comment:
1. if acct.Sku is really nil we'd better also skip it.

Consider just making Line40 to verify 'acct.Name != nil && acct.Location !=nil && acct.Sku != nil' , then remaining part could be simplified.

2. matchingAccountType check could come first to avoid unnecessary 'name & loc' assignment check

Reply:
good point, fixed